mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -267,9 +277,19 @@ class LinearRegression(override val uid: String)
     if (handlePersistence) instances.unpersist()
 
     val model = copyValues(new LinearRegressionModel(uid, weights, intercept))
+    // Handle possible missing or invalid prediction columns
+    val predictionColOpt = get(predictionCol).orElse(getDefault(predictionCol)).filter(_ != "")
```

Comment:
This code at the end of learning is getting complex enough that it would probably be worthwhile to create a helper method to avoid code duplication.  Would you mind doing that?  It'd be nice if it could cover evaluate() as well, to share code across these 3 places.